tools/clang/include/clang/Basic/Attr.td

```diff
@@ -1157,6 +1157,13 @@ def HLSLRayQueryObject : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def HLSLSubObject : InheritableAttr {
+  let Spellings = []; // No spellings!
+  let Subjects = SubjectList<[CXXRecord]>;
+  let Documentation = [Undocumented];
+  let Args = [UnsignedArgument<"SubObjKindUint">, DefaultIntArgument<"HitGroupType", -1>];
```

Comment:
Let's remove the default argument and make sure it's always a defined `DXIL::HitGroupType` enum value, which means this could also be unsigned.
```suggestion
  let Args = [UnsignedArgument<"SubObjKindUint">, UnsignedArgument<"HitGroupType">];
```